contracts/farm/Pool.sol

```diff
@@ -53,6 +53,7 @@ library Pool {
     bool onReferralBonus;
     uint256 totalReferralAmount; // deposited through referral
     FixedPointMath.uq192x64 accumulatedReferralWeight;
+    uint256 lockUpPeriodInSecs;
```

Comment:
I think the Data struct can be optimized by rearranging the variable definitions.  Currently it takes 11 storage slots but if we put the `IERC20 token, bool needVesting, bool onReferralBonus` at the bottom(or top), then it will take 9 slots, so will save 2*20k = 40k gas.